src/Product/WCProductAdapter.php

```diff
@@ -903,4 +912,166 @@ public function setTargetCountry( $targetCountry ) {
 		// product shipping information is also country based
 		$this->map_wc_product_shipping();
 	}
+
+	/**
+	 * Performs the attribute mapping.
+	 * This function applies rules setting values for the different attributes in the product.
+	 *
+	 * @param array $mapping_rules The set of rules to apply
+	 */
+	protected function map_attribute_mapping_rules( array $mapping_rules ) {
+		$attributes = [];
+
+		if ( empty( $mapping_rules ) ) {
+			return $this;
+		}
+
+		$this->set_product_category_ids();
```

Comment:
This at least won't call it for every mapping rule, but it's still not guaranteed that we need those categories. We've already called the function `map_product_categories` couldn't we just cache the ID's from there?

Also I noticed it calls `wc_get_product_cat_ids` which includes the hierarchy (fetching category ancestors), which is different to the way it's done in `set_product_category_ids`. I'm not sure how the category selection works but would it provide better matching?

For example if we have categories like:
```
Clothes
   > Shirts
   > Jackets
```

If the rule is set to apply to clothes, but the product is assigned to shirts, then we'll only have a match if we fetch categories including ancestors.